api/api.go

```diff
@@ -112,3 +125,342 @@ func (api *API) statsHandler(ctx router.HTTPContext) {
 		ctx.UnsupportedAPIMethod()
 	}
 }
+
+// ============================== HELPER FUNCTIONS ============================== //
+
+func postEntityHandler(ctx router.HTTPContext, c db.Connector) {
+	// Decode request body into entity var
+	var entity db.Entity
+	err := json.NewDecoder(ctx.Request.Body).Decode(&entity)
+	if err != nil {
+		ctx.APIError(router.ErrInternal, "Can't decode request body (error: %s)", err)
+		return
+	}
+
+	ConvertFloat64ToInt(entity)
+
+	for _, v := range entity {
+		if !validValueType(v) {
+			ctx.APIError(router.ErrBadRequest, "Invalid Value Type: %v. Value type must be string or int", reflect.TypeOf(v))
+		}
+	}
+
+	entities := []db.Entity{entity}
+	ids, err := c.CreateEntities(entities)
```

Comment:
Since `entities` isn't used elsewhere, you can just `c.CreateEntities([]db.Entity{entity})`.